bitcoin/script/miniscript.h

```diff
@@ -516,6 +532,26 @@ struct Node {
         return SanitizeType(ComputeType(fragment, x, y, z, sub_types, k, data.size(), subs.size(), keys.size()));
     }
 
+    //! Returns true if the node contains at least one duplicate key.
+    bool ContainsDuplicateKey() const {
+        auto upfn = [](const Node& node, Span<std::set<Key>> subs) -> std::optional<std::set<Key>> {
+            size_t keys_count = node.keys.size();
```

Comment:
Perhaps add `if (node.duplicate_key) return {};`. That way no work is done in recomputing these sets all the time if we already know there is a duplicate.

Reply:
Hmm but the boolean would be uninitialized for the current node. Isn't its value undefined?

Reply:
Ah, yes, that's a bad idea. For the current node, we obviously don't know yet. But for any child nodes we do.

So what about `if (&node != this && node.duplicate_key) return {};`.

Reply:
So there isn't any point in actually computing the set for child, right? We just use the `duplicate_key` boolean: if one messed up with the keys after the `Node` was created they were really looking for troubles..

Reply:
I don't follow.